### DISCUSSION TITLE
HTMLProofer Protocol-Relative URL and HTTPS Link Failures with Custom Subdomain

### DISCUSSION THREAD

**Discussion by DigitalDenCloud** [cotes2020/jekyll-theme-chirpy#1628] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

After deploying the Jekyll site on GitHub Pages with a custom domain (docs.digitalden.cloud) I encountered HTML validation failures during  the test site job, specifically with htmlproofer. 

Despite correctly setting up a CNAME record in Route 53 to point to digitalden3.github.io, and configuring chirpy to use digitalden3.github.io / docs.digitalden.cloud as its URL, I'm still facing issues with HTML checks related to protocol-relative URLs and HTTPS links. 

These checks are failing despite the site being accessible via the custom domain. The core issue seems to revolve around HTML validation errors that I'm unable to bypass or correct due to the way links are handled within your site or theme.

### Operations you have already tried

1. Setting up a CNAME record at GitHub for docs.digitalden.cloud pointing to digitalden3.github.io.
2. Created a CNAME record in Route 53, with docs pointing to digitalden3.github.io, to connect your custom domain.
3. Configured my config.yaml to use digitalden3.github.io and docs.digitaden.cloud as the URL.
4. Run htmlproofer with specific options (`--disable-external`, `--check-html`, `--allow-hash-href=true`, `--ignore-urls`) in an attempt to bypass the errors related to protocol-relative URLs and ensure that HTTPS links are validated correctly.
5. Used dig to confirm that docs.digitalden.cloud correctly resolves to the IP addresses of digitalden3.github.io.
6. Attempted to disable HTML checks that are causing validation failures, seeking a way to bypass these errors

### Anything else?

Simply commenting out the html proofer solve the problem, but I would love to know how I can solve this problem.         

bundle exec htmlproofer _site \
            \-\-disable-external=true \
            \-\-ignore-urls "/^http:\/\/127.0.0.1/,/^http:\/\/0.0.0.0/,/^http:\/\/localhost/"

Thank you in advance, and love this theme!

<img width="1067" alt="Screenshot 2024-03-23 at 18 10 37" src="https://github.com/cotes2020/jekyll-theme-chirpy/assets/98281161/c3b397d0-505c-419b-b53d-d11e4fd81cc7">

**Comment by cotes2020** [comment]
All that writing and no mention of which version you use.

**Reply by DigitalDenCloud** [reply]
hahahaa sorry im a newb, and first time posting.

Jekyll: 4.3.3
html-proofer: 4.4.3
Chirpy: 6.5.3

**Comment by kungfux** [comment]
It looks like it might be related to #1625 that was recently fixed.

I see test job has passed for you using the latest 6.5.5. Please confirm.

**Comment by DigitalDenCloud** [comment]
Sorry for the silly question, but just so I understand when issue #1625 was fixed, simply by re running my failed job it would be fixed on my end?

**Reply by kungfux** [reply]
Yes. The build is based on dependency defined in your [Gemfile](https://github.com/digitalden3/digitalden3.github.io/blob/main/Gemfile) as

```Gemfile
gem "jekyll-theme-chirpy", "~> 6.5", ">= 6.5.3"
```

that takes latest available patch version. You can read more [here](https://guides.rubygems.org/patterns/#pessimistic-version-constraint).
